### PR TITLE
Added classes for HW17 PetrolStation and ThreadSafeList

### PR DESCRIPTION
**Pull request by Dimidow**

### CODE REVIEW

File: HW17/src/ThreadSafeList.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class ThreadSafeList<T> {
```
Comment (serhii-tokranov) [inline HW17/src/ThreadSafeList.java]: The class must implement the List interface in a thread-safe manner. Methods add, get and remove must be implemented. The rest of the methods can be empty/not implemented (for example, throw OperationNotSupportedException).

Comment (Dimidow) [reply HW17/src/ThreadSafeList.java]: I see, thanks!

File: HW17/src/ThreadSafeList.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @Override
+    public Object[] toArray(Object[] a) {
+        return new Object[0];
+    }
+
+    @Override
+    public boolean add(Object o) {
+        synchronized (threadSafeList) {
```
Comment (serhii-tokranov) [inline HW17/src/ThreadSafeList.java:43]: I recommend using a standalone object as a lock. Good practice from the perspective of responsibility segregation (one thing = one functionality).

`private final Object lock = new Object()`

File: HW17/src/ThreadSafeList.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+
+public class ThreadSafeList<E> implements List<E> {
```
Comment (serhii-tokranov) [inline HW17/src/ThreadSafeList.java:7]: Also, you can try to extend the `AbstractList` class - fewer methods will need to be implemented.

**Pull request merged**